options/path.c

```diff
@@ -221,6 +221,16 @@ char *mp_get_user_path(void *talloc_ctx, struct mpv_global *global,
     return res;
 }
 
+char *mp_normalize_user_path(void *talloc_ctx, struct mpv_global *global,
+                             const char *path)
+{
+    assert(talloc_ctx && "mp_normalize_user_path requires talloc_ctx!");
+    char *expanded = mp_get_user_path(NULL, global, path);
```

Comment:
If talloc_ctx is required, why not use it here?

Reply:
It's only required because `mp_normalize_path` below requires it. I thought it was better to free the unneeded memory after the path is constructed.

Reply:
I've added a commit that fixes that as simplifies several calls of it as well.